### PR TITLE
Add task, attempt, question, and answer tables

### PR DESCRIPTION
**Pull request by codetheweb**
Closes https://github.com/sfbrigade/project-compass/issues/13 and https://github.com/sfbrigade/project-compass/issues/12

### CODE REVIEW

File: server/src/db/migrations/1_initial-migrations.sql [hunk 1]
```diff
@@ -1,6 +1,47 @@
...
+  task_attempt_id UUID PRIMARY KEY DEFAULT uuid_generate_v4(),
+  task_id UUID NOT NULL REFERENCES task(task_id),
+  student_id UUID NOT NULL REFERENCES student(student_id),
+  -- todo: add reference to the para who administered this attempt
+  created_at TIMESTAMP NOT NULL DEFAULT NOW()
+);
+
+CREATE DOMAIN task_attempt_question_answer_jsonb AS JSONB;
+
+CREATE TABLE task_attempt_question_answer (
```
Comment (codetheweb) [inline server/src/db/migrations/1_initial-migrations.sql:41]: not completely sold on this name, open to other ideas

Comment (tessathornberry) [reply server/src/db/migrations/1_initial-migrations.sql:48]: I was thinking "attempts" was sufficient for the table name, but "task_attempts" is probably more correct.

**Comment by amantri** [comment]
sorry - accidentally close the wrong pull request.

File: server/src/db/migrations/1_initial-migrations.sql [hunk 2]
```diff
@@ -13,3 +13,42 @@ CREATE TABLE "student" (
...
+  task_attempt_question_answer_id UUID PRIMARY KEY DEFAULT uuid_generate_v4(),
+  task_attempt_id UUID NOT NULL REFERENCES task_attempt(task_attempt_id),
+  task_question_id UUID NOT NULL REFERENCES task_question(task_question_id),
+  answer task_attempt_question_answer_jsonb NOT NULL,
```
Comment (amantri) [inline server/src/db/migrations/1_initial-migrations.sql:52]: Is it a postgres convention to suffix _jsonb here? We don't add the types to other fields.

**Pull request closed without merging**